packages/database/src/TupaiaDatabase.js

```diff
@@ -560,6 +560,12 @@ function buildQuery(connection, queryConfig, where = {}, options = {}) {
         }
       }
 
+      // Special case to handle CASE statements, otherwise they get interpreted as column names
+      const CASE_PATTERN = /^CASE/;
+      if (CASE_PATTERN.test(selector)) {
+        return { [alias]: connection.raw(selector) };
+      }
+
```

Comment:
We do seem to be growing a list of the special case handlers for custom column selectors. Might be nice to find a way for them to be handled as `raw` always, since they aren't at risk of sql injection. But that's for another ticket :)